test/ts/unit/simple_interest_terms_contract.ts

```diff
@@ -0,0 +1,237 @@
+import * as ABIDecoder from "abi-decoder";
+import * as chai from "chai";
+import * as _ from "lodash";
+import * as Units from "../test_utils/units";
+
+import {BigNumber} from "bignumber.js";
+
+import {RepaymentRouterContract} from "../../../types/generated/repayment_router";
+import {SimpleInterestTermsContractContract} from "../../../types/generated/simple_interest_terms_contract";
+import {MockDebtRegistryContract} from "../../../types/generated/mock_debt_registry";
+import {MockERC20TokenContract} from "../../../types/generated/mock_e_r_c20_token";
+import {MockTokenTransferProxyContract} from "../../../types/generated/mock_token_transfer_proxy";
+
+import {RepaymentRouterErrorCodes} from "../../../types/errors";
+
+import {BigNumberSetup} from "../test_utils/bignumber_setup";
+import ChaiSetup from "../test_utils/chai_setup";
+import {INVALID_OPCODE, REVERT_ERROR} from "../test_utils/constants";
+
+import {LogError, LogRepayment} from "../logs/repayment_router";
+
+import * as moment from "moment";
+
+// Set up Chai
+ChaiSetup.configure();
+const expect = chai.expect;
+
+// Configure BigNumber exponentiation
+BigNumberSetup.configure();
+
+const repaymentRouterContract = artifacts.require("RepaymentRouter");
+const simpleInterestTermsContract = artifacts.require("SimpleInterestTermsContract");
+
+contract("SimpleInterestTermsContract (Unit Tests)", async (ACCOUNTS) => {
+    let termsContract: SimpleInterestTermsContractContract;
+    let router: RepaymentRouterContract;
+    let mockToken: MockERC20TokenContract;
+    let mockRegistry: MockDebtRegistryContract;
+    let mockTokenTransferProxy: MockTokenTransferProxyContract;
+
+    const CONTRACT_OWNER = ACCOUNTS[0];
+    const PAYER = ACCOUNTS[1];
+    const BENEFICIARY = ACCOUNTS[2];
+    const ATTACKER = ACCOUNTS[3];
+
+    const TERMS_CONTRACT_PARAMETERS =
+        web3.sha3("any 32 byte hex value can represent the terms contract's parameters");
+    const ARBITRARY_AGREEMENT_ID =
+        web3.sha3("any 32 byte hex value can represent an agreement id");
+
+    const NULL_ADDRESS = "0x0000000000000000000000000000000000000000";
+
+    const TX_DEFAULTS = { from: CONTRACT_OWNER, gas: 4000000 };
+
+    before(async () => {
+        mockRegistry = await MockDebtRegistryContract.deployed(web3, TX_DEFAULTS);
+        mockToken = await MockERC20TokenContract.deployed(web3, TX_DEFAULTS);
+        mockTokenTransferProxy = await MockTokenTransferProxyContract.deployed(web3, TX_DEFAULTS);
+
+        const repaymentRouterTruffle = await repaymentRouterContract.new(
+          mockRegistry.address,
+          mockTokenTransferProxy.address
+        );
+
+        const termsContractTruffle = await simpleInterestTermsContract.new(
+            mockRegistry.address,
+            mockToken.address,
+            repaymentRouterTruffle.address,
+        );
+
```

Comment:
@saturnial you'll need to mock the payer's `balance` in `mockToken` and `allowance` in `mockToken` to the `mockTokenTransferProxy`, or else the `repay` function of the `RepaymentRouter` will fail across the board.  See [here](https://github.com/dharmaprotocol/charta/blob/78bacda84e284084f331c65100a09a06c6be892a/contracts/RepaymentRouter.sol#L92)

Reply:
Something like this?

```// The PAYER is endowed with a balance of 5 ether.
        await mockToken.mockBalanceOfFor.sendTransactionAsync(
            PAYER, Units.ether(5),
        );

        // The router is granted allowance to 3 ether from the PAYER.
        await mockToken.mockAllowanceFor.sendTransactionAsync(
            PAYER, router.address, Units.ether(3),
        );```

Reply:
Close! The allowance needs to be mocked for the `tokenTransferProxy`, given that `tokenTransferProxy` is the one calling `transferFrom` on `mockToken`.

Reply:
ah, so instead of granting the allowance to the `router`'s address, we grant it to the `mockTokenTransferProxy`'s address?

Reply:
Yessir